### ISSUE TITLE
Add error-reporting plugins

### ISSUE THREAD

**Issue by jamesarosen** [quirkey/sammy#75]
I know of two good ones: [Exceptional](https://github.com/contrast/exceptional-js/blob/master/public/exceptional.js) and [HopToad](http://hoptoadapp.com/javascripts/notifier.js).

Suggested approach: the plugin simply binds a function to the `"error"` event, which `app.error` triggers. That way they don't have to re-implement the default logic of that method.

My question is should this go in Sammy or in some new Sammy-contrib project?

**Comment by jamesarosen** [comment]
For HopToad:

```
this.bind('error', function(data) {
  setTimeout(function() {
      Hoptoad.notify({
        message : data.message,
        stack   : '()@' + STACK_INFO_HERE
      });
    }, 100);
});
```

For Exceptional:

```
this.bind('error', function(data) {
  setTimeout(function() {
      Exceptional.notify(data.message, window.location.href, LINE_HERE);
    }, 100);
});
```

Is there anything meaningful we could put in `STACK_INFO_HERE` or `LINE_HERE`?

**Comment by mrb** [comment]
I think a Sammy-contrib repo is in order for stuff like this.

**Comment by jamesarosen** [comment]
I'm totally for Sammy-contrib. Aaron: do you want to chime in on this one?

**Comment by quirkey** [comment]
For now I think we can keep this in the main repo. I think heading to 1.0 we'll have to break out in to a seperate repo with more plugins with the possibility of more specific 'less core' plugins. I would definitely use the hoptoad one, though!

**Comment by jamesarosen** [comment]
10-4. I'll put them in here for now. I believe they're both quite small.

**Comment by jamesarosen** [comment]
I'm having trouble testing these. I've defined a double of `Hoptoad.notify` and created a route that throws an error. Every time I try to load the page, though, it goes into an endless refresh loop.

**Comment by jamesarosen** [comment]
Resolved in pull request #78.

**Comment by quirkey** [comment]
Pulled. Closing
